src/Symfony/Bridge/Twig/Mime/Renderer.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Twig\Mime;
+
+use League\HTMLToMarkdown\HtmlConverter;
+use Twig\Environment;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @experimental in 4.3
+ */
+final class Renderer
+{
+    private $twig;
+    private $context;
+    private $converter;
+
+    public function __construct(Environment $twig, array $context = [])
+    {
+        $this->twig = $twig;
+        $this->context = $context;
+        if (class_exists(HtmlConverter::class)) {
+            $this->converter = new HtmlConverter([
```

Comment:
How could we provide autodiscovery for this?
A "suggest" entry in composer.json maybe, but that's barely visible.
I don't have a better idea. (making it a requirement with runtime suggestion for the package?)

Reply:
That seems like a good place to start - throw an exception when/if it's used (in `convertHtmlToText()`). So, if you set the text & html body, you'll never need it and never get an exception about missing it. But if you do, we say:

> Cannot automatically convert HTML body to a text body. Run "composer require league/html-to-markdown"
> or set the text content explicitly.

Reply:
I like your suggestion @weaverryan, can you create a PR for it?